src/debug/ee/debugger.cpp

```diff
@@ -2897,70 +2894,82 @@ DebuggerJitInfo *Debugger::GetJitInfoWorker(MethodDesc *fd, const BYTE *pbAddr,
         return NULL;
     }
 
-
+#ifdef DACCESS_COMPILE
+    dji = dmi->GetLatestJitInfo();
```

Comment:
I'm not sure if we have good enough test coverage to risk changing this method. In particular my concerns would be EnC (I don't think we have any tests enabled in CoreCLR at this point) and any performance impact to the stack trace scenario (we have a hole there in a perf testing).

FWIW looking at it I didn't see anything you wrote that appears to cause an issue, I'm just being cautious. I think it would be worthwhile saving or commenting instead because it would be nice to make this method less quirky. It also suggests that we may have 2nd bug when trying to use System.Diagnostics.StackTrace on tiered code which I think also comes through here even if debugging is disabled.

Instead of changing this method I'd suggest doing a less invasive fix where you make GetILToNativeMapping call straight over to FindOrCreateInitAndAddJitInfo, skipping GetJitInfoFromAddr, GetJitInfo, and GetJitInfoWorker.